pkg/config/config_internal_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+type file struct {
+	filename string
+	content  []byte
+}
+
+//nolint:funlen
+func TestBodyFromPath(t *testing.T) {
+	tests := []struct {
+		desc  string
+		files []file
+	}{
+		{
+			desc: "Single file",
+			files: []file{
+				{
+					filename: "cluster.lokocfg",
+					content: []byte(`cluster "aws" {
+  asset_dir        = pathexpand("./assets")
+  cluster_name     = var.cluster_name
+  controller_count = 1
+  dns_zone         = "example.com"
+  dns_zone_id      = "fake"
+  ssh_pubkeys      = ["stuff"]  
+  worker_pool "my-wp-name" {
+    count         = 3
+    instance_type = "fake"
+    ssh_pubkeys   = ["stuff"]
+  }
+}
+
+component "foo" {}
+component "bar" {}`),
+				},
+			},
+		},
+		{
+			desc: "Multiple files",
+			files: []file{
+				{
+					filename: "cluster.lokocfg",
+					content: []byte(`cluster "aws" {
+  asset_dir        = pathexpand("./assets")
+  cluster_name     = var.cluster_name
+  controller_count = 1
+  dns_zone         = "example.com"
+  dns_zone_id      = "fake"
+  ssh_pubkeys      = ["stuff"]  
+  worker_pool "my-wp-name" {
+    count         = 3
+    instance_type = "fake"
+    ssh_pubkeys   = ["stuff"]
+  }
+}`),
+				},
+				{
+					filename: "components.lokocfg",
+					content: []byte(`component "foo" {}
+component "bar" {}`),
+				},
+			},
+		},
+		{
+			desc: "Unrelated files in config dir",
+			files: []file{
+				{
+					filename: "cluster.lokocfg",
+					content: []byte(`cluster "aws" {
+  asset_dir        = pathexpand("./assets")
+  cluster_name     = var.cluster_name
+  controller_count = 1
+  dns_zone         = "example.com"
+  dns_zone_id      = "fake"
+  ssh_pubkeys      = ["stuff"]  
+  worker_pool "my-wp-name" {
+    count         = 3
+    instance_type = "fake"
+    ssh_pubkeys   = ["stuff"]
+  }
+}`),
+				},
+				{
+					filename: "foo.txt",
```

Comment:
We already have `desc: "Unrelated files in config dir"`. I can make it "Unrelated files in config dir should be ignored" but I find it kind of obvious since the files are unrelated...

Reply:
In this specific test function, there are only two possible outcomes for a test case: "errors" and "no errors". I thought it's clear what when we don't have e.g. `wantDiags` there shouldn't be any errors.

Reply:
Yeah, I read the tests again and they seem all right, resolving.